exporters/sender/okhttp/src/test/java/io/opentelemetry/exporter/sender/okhttp/internal/OkHttpSenderTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.sender.okhttp.internal;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.opentelemetry.exporter.internal.InstrumentationUtil;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+abstract class OkHttpSenderTest<T> {
+
+  @BeforeEach
+  void setUp() {
+    OkHttpUtil.propagateContextInDispatcher = true;
+  }
+
+  @AfterEach
+  void tearDown() {
+    OkHttpUtil.propagateContextInDispatcher = false;
+  }
+
+  @Test
+  void testSuppressInstrumentation() throws InterruptedException {
+    CountDownLatch latch = new CountDownLatch(1);
+    AtomicBoolean suppressInstrumentation = new AtomicBoolean(false);
+
+    Runnable onSuccess = Assertions::fail;
+    Runnable onFailure =
```

Comment:
onFailure is always called because the `https://none` always fails to connect?

Reply:
Yes, I get a "Failed to connect to [host]" error. At first, I tried to also test the `onSuccess` case by creating a [MockWebServer](https://github.com/square/okhttp/tree/master/mockwebserver), which worked well but only when testing `OkHttpHttpSender`. But it would always fail for `OkHttpGrpcSender` (I think it might have something to do with this [line](https://github.com/open-telemetry/opentelemetry-java/blob/7fd46f058350f35cb222eec1f76153b3751121b8/exporters/sender/okhttp/src/main/java/io/opentelemetry/exporter/sender/okhttp/internal/OkHttpGrpcSender.java#L91) but I'm not entirely sure, it's probably an issue in the MockWebServer lib). So in the end I just left the `onFailure` test which worked well for both senders.